app/services/ems_storage_dashboard_service.rb

```diff
@@ -90,4 +95,36 @@ def heatmaps
       :title         => 'Used capacity [%]'
     }
   end
+
+  def agg_events
+    event_hash = {}
+    event_list = []
+    events = EmsEvent.where("ems_id=#{@ems_id}")
+
+    events.each do |event|
```

Comment:
You could use the association here since it looks like you have `@ems`

```suggestion
    @ems.ems_events.each do |event|
```